src/main/java/com/easypost/http/Requestor.java

```diff
@@ -553,7 +560,20 @@ private static <T> T httpRequest(final RequestMethod method, final String url, f
                     break;
             }
         }
-
+        Instant requestTimestamp = Instant.now();
+        UUID requestUuid = UUID.randomUUID();
+        Map<String, String> headers = new HashMap<String, String>();
+        headers = generateHeaders(client.getApiKey());
+
+        HashMap<String, Object> requestBodyForHook = new HashMap<String, Object>();
+        requestBodyForHook.put("headers", headers);
```

Comment:
Let's not use the Pythonic kwargs approach here for Java. We should be calling the function with six explicit parameters rather than one dictionary. Users can better account for exactly what parameters will be passed into their callback function then, instead of needing to parse a dictionary.

Reply:
Having six explicit parameters will require users to have those parameters in the custom function they are passing in. If we don't like the hashmap, using an object with getters might be better

Reply:
An object could also work. I don't think the parameter order is necessarily a bad thing, would require them to know the method signature (this is already what we do with Ruby): https://github.com/EasyPost/easypost-ruby#custom-connections

Reply:
I think explicit parameter order is the worst approach. Users would need to know all the parameters and the correct order, if we add additional request/response values for the hooks in the future, it would be a breaking change for the users whereas using hashmap and object will not.

Reply:
Good argument. I'd be in favor of an object then.